### PR TITLE
Update criteo_processor.py

### PR DESCRIPTION
**Pull request by soma2000-lang**
## Relevant Issue
#164 

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/NimbleEdge/EnvisEdge/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/NimbleEdge/EnvisEdge/blob/main/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [NimbleEdge Styleguide](https://numpydoc.readthedocs.io/en/latest/format.html)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/NimbleEdge/EnvisEdge/labels)
- [x] My changes are covered by tests

### CODE REVIEW

**Comment by soma2000-lang** [comment]
@divyanshugit  Please look into it

File: datasets/criteo/criteo_processor.py [hunk 1]
```diff
@@ -16,30 +16,81 @@
...
+
+
+    def __init__(         
             self,
             datafile,
             output_file,
             max_ind_range=0,
             sub_sample_rate=0.0,
-            randomize="day",
+            randomize="day",#we are randomizing the dates for better results
```
Comment (divyanshugit) [inline datasets/criteo/criteo_processor.py:65]: We are already using Numpy Style guidelines to comment on the code. So, we don't have to use this.

File: datasets/criteo/criteo_processor.py [hunk 2]
```diff
@@ -16,30 +16,81 @@
...
+     """
+        
         self.datafile = datafile
         self.output_file = output_file
         lstr = datafile.split("/")
         self.d_path = "/".join(lstr[0:-1]) + "/"
         self.d_file = lstr[-1].split(".")[0]
         self.npzfile = self.d_path + (self.d_file + "_day")
-        self.trafile = self.d_path + (self.d_file + "_fea")
+        self.trafile = self.d_path + (self.d_file + "_fea")#The tarfile makes it possible to read and write tar archives
```
Comment (divyanshugit) [inline datasets/criteo/criteo_processor.py:88]: Ditto

File: datasets/criteo/criteo_processor.py [hunk 3]
```diff
@@ -16,30 +16,81 @@
...
         self.total_file = self.d_path + self.d_file + "_day_count.npz"
 
         self.dataset_multiprocessing = dataset_multiprocessing
         self.sub_sample_rate = sub_sample_rate
-        self.days = 7
+        self.days = 7 #There are 7 days in a week
```
Comment (divyanshugit) [inline datasets/criteo/criteo_processor.py:93]: ditto

File: datasets/criteo/criteo_processor.py [hunk 4]
```diff
@@ -48,9 +99,9 @@ def __init__(
         # tot_fea = tad_fea + spa_fea
         self.randomize = randomize
         self.max_ind_range = max_ind_range
-        self.clear_items()
+        self.clear_items() #clear() method removes all items from the list.
```
Comment (divyanshugit) [inline datasets/criteo/criteo_processor.py:102]: ditto

File: datasets/criteo/criteo_processor.py [hunk 5]
```diff
@@ -48,9 +99,9 @@ def __init__(
         # tot_fea = tad_fea + spa_fea
         self.randomize = randomize
         self.max_ind_range = max_ind_range
-        self.clear_items()
+        self.clear_items() #clear() method removes all items from the list. 
 
-    @staticmethod
+    @staticmethod # plain functions but we can call them from an instance or the class"""
```
Comment (divyanshugit) [inline datasets/criteo/criteo_processor.py:104]: ditto

File: datasets/criteo/criteo_processor.py [hunk 6]
```diff
@@ -74,7 +143,7 @@ def _process_one_file(
             if sub_sample_rate == 0.0:
                 rand_u = 1.0
             else:
-                rand_u = np.random.uniform(
+                rand_u = np.random.uniform(                     #Draw samples from a uniform distribution.
```
Comment (divyanshugit) [inline datasets/criteo/criteo_processor.py:146]: ditto

File: datasets/criteo/criteo_processor.py [hunk 7]
```diff
@@ -116,10 +185,10 @@ def _process_one_file(
                 np.savez_compressed(
                     filename_s,
                     X_int=X_int[0:i, :],
-                    X_cat_t=np.transpose(X_cat[0:i, :]),
+                    X_cat_t=np.transpose(X_cat[0:i, :]),# permute the axes of the array
```
Comment (divyanshugit) [inline datasets/criteo/criteo_processor.py:188]: ditto

File: datasets/criteo/criteo_processor.py [hunk 8]
```diff
@@ -116,10 +185,10 @@ def _process_one_file(
...
                     y=y[0:i],
                 )
-                print("\nSaved " + npzfile + "_{0}.npz!".format(split))
+                print("\nSaved " + npzfile + "_{0}.npz!".format(split))#file saved in npz format
```
Comment (divyanshugit) [inline datasets/criteo/criteo_processor.py:191]: ditto

File: datasets/criteo/criteo_processor.py [hunk 9]
```diff
@@ -167,7 +252,7 @@ def process_data(self):
         counts = np.zeros(26, dtype=np.int32)
         # create dictionaries
         for j in range(26):
-            for i, x in enumerate(convertDicts[j]):
+            for i, x in enumerate(convertDicts[j]):#enumerate adds a counter to an iterable and returns it in a form of enumerating object
```
Comment (divyanshugit) [inline datasets/criteo/criteo_processor.py:255]: ditto

File: datasets/criteo/criteo_processor.py [hunk 10]
```diff
@@ -200,10 +285,16 @@ def process_data(self):
                 CriteoDataProcessor.processCriteoAdData(
                     self.npzfile, i, self.days, convertDicts)
 
-        return self.concat_data(self.output_file)
+        return self.concat_data(self.output_file) #pandas.concat()
```
Comment (divyanshugit) [inline datasets/criteo/criteo_processor.py:288]: ditto

**Comment by soma2000-lang** [comment]
@divyanshugit  ok I am immediately removing them

**Comment by divyanshugit** [comment]
Mention those things in the inline comments itself

**Comment by soma2000-lang** [comment]
@divyanshugit  Apart from these is there  any further changes to be done.

File: datasets/criteo/criteo_processor.py [hunk 11]
```diff
@@ -16,30 +16,81 @@
...
 # split (bool) : to split into train, test, validation data-sets
 
-
-@registry.load('dataset', 'kaggle')
+@registry.load('dset_proc', 'kaggle')#We decorate the class definitions with registry.load and create a yaml configuration file of all the arguments to pass
 class CriteoDataProcessor:
-    def __init__(
+
+
+
```
Comment (divyanshugit) [inline datasets/criteo/criteo_processor.py:23]: Can you please remove these types of unnecessary blank lines?

Comment (soma2000-lang) [reply datasets/criteo/criteo_processor.py:23]: Ok

**Pull request closed without merging**